web/src/people.ts

```diff
@@ -14,15 +14,65 @@ import * as settings_config from "./settings_config";
 import * as settings_data from "./settings_data";
 import * as util from "./util";
 
-let people_dict;
-let people_by_name_dict;
-let people_by_user_id_dict;
-let active_user_dict;
-let non_active_user_dict;
-let cross_realm_dict;
-let pm_recipient_count_dict;
-let duplicate_full_name_data;
-let my_user_id;
+export type ProfileData = {
+    value: string;
+    rendered_value: string;
+};
+
+export type User = {
+    user_id: number;
+    delivery_email?: string;
+    email: string;
+    full_name: string;
+    date_joined: string;
+    is_active: boolean;
+    is_owner: boolean;
+    is_admin: boolean;
+    is_guest: boolean;
+    is_billing_admin: boolean;
+    is_bot: boolean;
+    bot_type?: number;
+    bot_owner_id?: number;
+    role: number;
+    timezone: string;
+    avatar_url?: string;
+    avatar_version: number;
+    profile_data: Record<number, ProfileData>;
+};
```

Comment:
It'd be good to have a comment on this type for whether it is exactly the type for `User` we'll get from the server via the API, or whether any processing has happened. (Or if both appear in the file, we might want a separate `APIUser` type that is exactly the one the API gives and comments explaining the difference.